libkbfs/block_types.go

```diff
@@ -175,11 +201,20 @@ func (fb *FileBlock) Set(other Block, codec kbfscodec.Codec) {
 	if err != nil {
 		panic("Unable to DirBlock.Set")
 	}
-	*fb = *copy
+	fb.cacheMtx.Lock()
```

Comment:
Same comments here as in `DirBlock.Set()`.

However, does the cache lock now apply to `FileBlock.Hash` too?  Could use a comment in the `FileBlock` struct about that.